x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/InstanceType.java

```diff
@@ -72,6 +103,7 @@ public static InstanceType fromHostSource(Map<String, Object> source) {
         // We only support AWS for 8.12, but plan for GCP and Azure later.
         // "gcp": check 'gce.instance.name' or 'gce.instance.name' to extract the instanceType
         // "azure": extract the instanceType
+        String instanceType = "";
         if ("aws".equals(provider)) {
```

Comment:
It is a theoretical case where for some reason `ec2.placement.region` isn't available. This can e.g. happen when the host agent doesn't have permissions to access the AWS API.
That's why I hesitated to remove it in the first place, but it makes sense because `ec2.instance_type` won't be available in this case either.